docs/porting/porting_full_process/integration tests.md

```diff
@@ -0,0 +1,65 @@
+| `net-multi` | Network multithreaded test for 1, 2 and 3 download threads with 1kb receive buffer size. |
+| `stress-net-fs` | Network and file system single and multithreaded tests:<ul><li>memory allocation - 10k, 20k, 40k, 60k</li><li>1 thread (sequential) - 1 download (1kb buffer), 1 file thread (1kb buffer)</li><li>2 parallel threads - 1 download, 1 file thread (1kb buffer)</li><li>3 parallel threads - 1 download, 2 file (256 bytes, 1 kb buffer)</li><li>4 parallel threads - 1 download, 3 file (1 byte, 256 bytes, 1kb buffer)</li></ul> |
+
+## Execute tests
+
+The integration test requires you to configure both the connectivity and storage devices.
+
+If a platform has both of them enabled by default (in `targets.json`), then you can compile the tests and run them from inside the `mbed-os` directory: 
+
+    ```
+    $ mbed test -t <toolchain> -m <platform> -n *integration-* -DINTEGRATION_TESTS -v
+    ```
+    
+If a platform doesn't have default storage or connectivity device and may require extra shield to run the integration test. In this case, you can pass a `target_extended.json` file in the command-line to configure the target:
+
+   ```
+   $ mbed test -t <toolchain> -m <platform> -n *integration-* -DINTEGRATION_TESTS --app-config TESTS\integration\COMMON\target_extended.json -v
+   ```
+   
+A preconfiguration file is defined in the COMMON folder, and that file covers some of the platform connectivity and storage combinations. However, this preconfiguration file might contain some compoments drivers not included in `mbed-os` (for example, `WIFI_WIZFI310` and `NUSD`). You must add those drivers manually. If you have your own platform or want to define your own configuration, please follow the [configuration guide](#tests-configuration).
+
+## Tests configuration
+
+If required, edit the `target_extended.json` file, and create a new entry under `target_overrides` with the target name for your device:
+
+- **Connectivity** - Specify the default connectivity type for your target. It's essential with targets that lack default connectivity set in `targets.json` or for targets that support multiple connectivity options. For example:
+  
+   ```
+            "target.network-default-interface-type" : "ETHERNET",
+   ```
+   
+   The possible options are `ETHERNET`, `WIFI` and `CELLULAR`.
+   
+   Depending on connectivity type, you might have to specify more configuration options. For an example, please see the defined `CELLULAR` targets in `mbed_app.json`.
+
+- **Storage** - Specify the storage block device type, which dynamically adds the block device driver you specified at compile time. For example:
+
+   ```
+            "target.components_add" : ["SD"],
+   ```
+
+   Valid options are `SD`, `SPIF`, `QSPIF` and `FLASHIAP` (not recommended). For more available options, please see the [block device components](https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice).
```

Comment:
Why isn't FLASHIAP recommended? Is it the only one of the group not recommended?

Reply:
It's not recommended because of the risks of wearing out the flash memory of the device.
We should probably remove this statement as afaik we're not commenting on the possibility of its usage.